C++/include/algorithm/dynamic_programming/README.md

```diff
@@ -44,3 +50,197 @@ Time               | Space
+  Type: `long long`
+
+  Represents the total number of items given.
+
+- #### `weights`
+  Type: `vector<long long>`
+
+  vector of the weights for the items given.
+
+- #### `values`
+  Type: `vector<long long>`
+
+  vector of the values for the items given.
+
+### Output
+- #### `maxItems[numberOfItems][capacity]`
+  Type: `int (bottom rightmost cell of 2D Matrix of type vector<vector<long long>>)`
+
+  After filling the matrix the bottom rightmost cell of the of the matrix will contain maximum total value in the knapsack.  
+
+### Complexity
+
+Time               | Space
+-------------------|-------------------
+_O(N*C)_ | _O(N*C)_
+
+where N is the number of items and C is the capacity of the knapsack.
+
+## 4. Kadane
+
+This algorithm is used for finding the contiguous subarray within a one-dimensional array of integers which has the largest sum.
+
+### Input
+- #### `values`
+  Type: `vector<int>`
+
+  Vector of integers for which we have to find the contiguous subarray having the largest sum.
+
+### Output
+- #### `maxSum, start, end`
+  Type: `tuple<int, size_t, size_t>`
+
+  Output is a tuple of maximum sum, start index of the contiguous array and the last index of the contiguous array. Note: `size_t` is an unsigned integral data type.  
+
+### Complexity
+
+Time               | Space
+-------------------|-------------------
+_O(N)_ | _O(1)_
+
+where N is the number of elements in the original array.
+
+## 5. Longest Decreasing Subsequence
+
+This algorithm finds the longest decreasing subsequence in a sequence of numbers.
+
+### Input
+- #### `input`
+  Type: `vector<int>`
+
+  Vector array of inputs for which we have to find longest decreasing subsequence.
+
+- #### `to_show_state`
+  Type: `const bool`
+
+  Initially it is set to false, if no second argument is provided in the function. if this is true subsequence is printed out by the function.
+
+### Output
+- #### `max`
+  Type: `int`
+
+  Represents the largest length of the longest decreasing subsequence.  
+
+### Complexity
+
+Time               | Space
+-------------------|-------------------
+_O(n log(n))_ | _O(n)_
+
+where n is the number of elements in the initial list.
+
+## 6. Rod Cutting
+
+Given a rod of length N units and an array of prices that contains prices of all pieces of size smaller than N. Determine the maximum value obtainable by cutting up the rod and selling the pieces.
+
+### Input
+- #### `costs`
+  Type: `vector<std::size_t>`
+
+  Vector of costs of all pieces of size smaller than N.
+
+### Output
+- #### `cost`
+  Type: `size_t (size_t is an unsigned integral data type)`
+
+  Represents the maximum value obtainable by cutting up the rod and selling the pieces. 
+
+### Complexity
+
+Time               | Space
+-------------------|-------------------
+_O(N<sup>2</sup>)_ | _O(N)_
+
+where N is the length of the rod.
+
+## 7. Weighted Activity Selection
+
+Given N activities where every activity is represented by following three properties:
```

Comment:
Given N activities where every activity is represented by following three properties:

1. Start time
2. Finish time
3. Weight (or value)

find the maximum weight subset of activities such that no two activities overlap.